mesonbuild/dependencies.py

```diff
@@ -119,8 +119,10 @@ def __init__(self, name, environment, kwargs):
                 if self.required:
                     raise DependencyException('Pkg-config binary missing from cross file')
             else:
-                self.pkgbin = environment.cross_info.config['binaries']['pkgconfig']
-                PkgConfigDependency.class_pkgbin = self.pkgbin
+                potential_pkgbin = environment.cross_info.config['binaries'].get('pkgconfig', 'non_existing_binary')
+                if shutil.which('potential_pkgbin'):
```

Comment:
I did test it but we apparently don't have a test that cross compiles and requires a pkg-config dependency. Which is good because Debian seems to have removed cross-pkg-config binaries so these tests would have failed. Fixed, thanks for spotting this.